pkg/auth/auth.go

```diff
@@ -43,6 +44,41 @@ func CheckAuthFile(authfile string) error {
 	return nil
 }
 
+// Mirrors --authfile to a tmpfile if --authfile points to a
+// file descriptor instead of actual file on filesystem
+// reason is file descriptor could vanish mid-session if consumed
+// once, check https://github.com/containers/buildah/issues/3070
+func CreateTempAuthFileIfNeeded(authfile string) string {
+	if strings.HasPrefix(authfile, "/dev/fd") {
```

Comment:
The function would be simpler by inverting the logic and `return authfile` early. We could spare a big scope.